test/unit-tests/MultiMessageSender.t.sol

```diff
@@ -0,0 +1,375 @@
+        vm.expectRevert(Error.ZERO_CHAIN_ID.selector);
+        sender.remoteCall(0, address(42), bytes("42"));
+    }
+
+    /// @dev cannot call with target address of 0
+    function test_remote_call_zero_target_address() public {
+        vm.startPrank(caller);
+
+        vm.expectRevert(Error.INVALID_TARGET.selector);
+        sender.remoteCall(DST_CHAIN_ID, address(0), bytes("42"));
+    }
+
+    /// @dev cannot call with receiver address of 0
+    function test_remote_call_zero_receiver_address() public {
+        vm.startPrank(caller);
+
+        MultiMessageSender dummySender = new MultiMessageSender(address(new ZeroAddressReceiverGac(caller)));
+
+        vm.expectRevert(Error.ZERO_RECEIVER_ADAPTER.selector);
+        dummySender.remoteCall(DST_CHAIN_ID, address(42), bytes("42"));
+    }
+
+    /// @dev cannot call with no sender adapter
+    function test_remote_call_no_sender_adapter_found() public {
+        vm.startPrank(owner);
+
+        // Remove both adapters
+        address[] memory senderAdapters = new address[](2);
+        senderAdapters[0] = wormholeAdapterAddr;
+        senderAdapters[1] = axelarAdapterAddr;
+
+        sender.removeSenderAdapters(senderAdapters);
+
+        vm.startPrank(caller);
+
+        vm.expectRevert(Error.NO_SENDER_ADAPTER_FOUND.selector);
+        sender.remoteCall(DST_CHAIN_ID, address(42), bytes("42"));
+    }
+
+    /// @dev should proceed with the call despite one failing adapter, emitting an error message
+    function test_remote_call_failing_adapter() public {
+        vm.startPrank(owner);
+
+        // Add failing adapter
+        address[] memory addedSenderAdapters = new address[](1);
+        address failingAdapterAddr = address(new FailingSenderAdapter());
+        addedSenderAdapters[0] = failingAdapterAddr;
+        sender.addSenderAdapters(addedSenderAdapters);
+
+        vm.startPrank(caller);
+
+        address[] memory senderAdapters = new address[](3);
+        senderAdapters[0] = wormholeAdapterAddr;
+        senderAdapters[1] = axelarAdapterAddr;
+        senderAdapters[2] = failingAdapterAddr;
+
+        bool[] memory adapterSuccess = new bool[](3);
+        adapterSuccess[0] = true;
+        adapterSuccess[1] = true;
+        adapterSuccess[2] = false;
+
+        uint256 expiration = block.timestamp + gac.getMsgExpiryTime();
+
+        MessageLibrary.Message memory message = MessageLibrary.Message({
+            srcChainId: SRC_CHAIN_ID,
+            dstChainId: DST_CHAIN_ID,
+            target: address(42),
+            nonce: 1,
+            callData: bytes("42"),
+            expiration: expiration
+        });
+        bytes32 msgId = MessageLibrary.computeMsgId(message);
+
+        uint256 fee = sender.estimateTotalMessageFee(DST_CHAIN_ID, receiver, address(42), bytes("42"));
+
+        vm.expectEmit(true, true, true, true, address(sender));
+        emit ErrorSendMessage(failingAdapterAddr, message);
+
+        vm.expectEmit(true, true, true, true, address(sender));
```

Comment:
`MultiMessageMsgSent` event doesn't seem to have any indexed params. Can we index `msgId`, `dstChainId` and `target`

Reply:
Done.